src/extensions/scratch3_text2speech/index.js

```diff
@@ -722,46 +722,46 @@ class Scratch3Text2SpeechBlocks {
         path += `&text=${encodeURIComponent(words.substring(0, 128))}`;
 
         // Perform HTTP request to get audio file
-        return new Promise(resolve => {
-            nets({
-                url: path,
-                timeout: SERVER_TIMEOUT
-            }, (err, res, body) => {
-                if (err) {
-                    log.warn(err);
-                    return resolve();
-                }
-
-                if (res.statusCode !== 200) {
-                    log.warn(res.statusCode);
-                    return resolve();
+        return fetchWithTimeout(path, {}, SERVER_TIMEOUT)
+            .then(res => {
+                if (res.status !== 200) {
+                    throw new Error(`HTTP ${res.status} error reaching translation service`);
                 }
 
+                return res.arrayBuffer();
+            })
+            .then(buffer => {
                 // Play the sound
                 const sound = {
                     data: {
-                        buffer: body.buffer
+                        buffer: buffer
                     }
                 };
-                this.runtime.audioEngine.decodeSoundPlayer(sound).then(soundPlayer => {
-                    this._soundPlayers.set(soundPlayer.id, soundPlayer);
+                return this.runtime.audioEngine.decodeSoundPlayer(sound);
+            })
+            .then(soundPlayer => {
+                this._soundPlayers.set(soundPlayer.id, soundPlayer);
 
-                    soundPlayer.setPlaybackRate(playbackRate);
+                soundPlayer.setPlaybackRate(playbackRate);
 
-                    // Increase the volume
-                    const engine = this.runtime.audioEngine;
-                    const chain = engine.createEffectChain();
-                    chain.set('volume', SPEECH_VOLUME);
-                    soundPlayer.connect(chain);
+                // Increase the volume
+                const engine = this.runtime.audioEngine;
+                const chain = engine.createEffectChain();
+                chain.set('volume', SPEECH_VOLUME);
+                soundPlayer.connect(chain);
 
-                    soundPlayer.play();
+                soundPlayer.play();
+                return new Promise(resolve => {
                     soundPlayer.on('stop', () => {
                         this._soundPlayers.delete(soundPlayer.id);
                         resolve();
                     });
                 });
+            })
+            .catch(err => {
+                log.warn(err);
+                return;
```

Comment:
This `return;` seems unnecessary... is it there for lint purposes?

Reply:
I think it was, but it looks like linting still passes without it!